Automattic/jetpack
# CRM: Resolves 2880 - clean up date pickers in CRM

CRM: Resolves Automattic/zero-bs-crm#2880 - clean up date pickers in CRM

## Proposed changes:

The initial issue was to clean up some duplicate JS code used between Client Portal and Woo My Account. The code in question was directly related to the third-party daterangepicker library, so I went ahead and bypassed it with native functions instead. Doing so also affected custom fields of type `date` and quote date fields (due to tightly-wound code).

Note we currently store date fields, adjusted to the UTC timezone, stored in UTS. I adjusted the new date/time functions to support this, though ideally we'd store dates in `YYYY-MM-DD` format instead: pbhBOz-1s2-p2#comment-2188

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






Everything should work as before, with native date pickers. There's less JS needed on the Client Portal and My Account.

1. Create a contact custom field of type `date`.
2. Ensure viewing/modifying the field works as before
    * Contact view page
    * Contact edit page
    * Client Portal as contact
    * Woo My Account as contact (note that you'll need to enable the field in WooSync settings: `/wp-admin/admin.php?page=zerobscrm-plugin-settings&tab=woosync`
3. Make sure quote dates save/display as before.
